### DISCUSSION TITLE
Nested component loses focus on change validation

### DISCUSSION THREAD

**Discussion by hernanif1** [react-hook-form/react-hook-form#7530] [Q&A]
1. Nested components I mean components inside components like this:
```tsx
const Order = ()=> {
   const Form = ()=> ( .... )
   return <Form/>
}
```

2. Check the link with my example - https://codesandbox.io/s/react-hook-form-v7-form-context-forked-qzm8h?file=/src/index.js

```tsx
function App() {
  const {
    formState: { errors },
    handleSubmit,
    register
  } = useForm({
    resolver: yupResolver(schema)
  });

  const Form = () => (
    <form onSubmit={handleSubmit((data) => console.log(data))}>
      <input {...register("fullname")} />
      {errors.fullname?.message && <p>{errors.fullname?.message}</p>}
      <button type="submit">send</button>
    </form>
  );
  return <Form />;
}
```

3. Press enter or the click at the button first
4. Start typing and you will see when the validation changes (valid or not) the focus will be removed.
5. I tried to 
 - check all open issues
 - read the docs 
 - implement Provider / Controller 

6. It seems that I'm missing some important concept from React-hook-form that's why I didn't open an issue but a question here.

Thanks for all your support 🙏

**Comment by Moshyfawn** [comment]
It's more of a React nuance than RHF, I believe. Defining you `Form` component inside a functional component with a state means that the `Form` component will be re-mounted every re-render due to the form state changes. In your case, it's `errors` state that changes: you can see the fullname input doesn't get focused on error after form submission, because it's `ref` is different after the errors state changes from
```
{}
```
to
```
 fullname: {
   message: 'fullname must be at least 3 characters',
   type: 'min',
   ref: <input name='fullname'></input>,
 },
```
due to the substitution validation error.
Same happens when the validation error gets fixed.

**Reply by guplem** [reply]
Then, isn't there a way of fixing this behavior?

**Reply by Moshyfawn** [reply]
Yes, to move the component out of render. You can memo it as well, but it's a waste of computing power, imo

**Comment by Phibee** [comment]
Hi @hernanif1 , please checked the link below. I was able to fixed you issue with regards to lossing focus. Hope it helps. 

Code:
https://codesandbox.io/s/react-hook-form-v7-form-context-forked-1126x6?file=/src/index.js
